nservicebus/hosting/cloud-services-host/shared-hosting.md

```diff
@@ -50,59 +50,33 @@ In this worker role one needs to reference the assembly that contains the Azure
 
 To integrate the NServiceBus dynamic host into the worker role entry point, all one needs to do is to create a new instance of `NServiceBusRoleEntrypoint` and call it's `Start` and `Stop` methods in the appropriate `RoleEntryPoint` override.
 
-```
-public class WorkerRole : RoleEntryPoint
-{
-	NServiceBusRoleEntrypoint nsb = new NServiceBusRoleEntrypoint();
-	
-	public override bool OnStart()
-	{
-	    nsb.Start();	
-	    return base.OnStart();
-	}
-	
-	public override void OnStop()
-	{
-	    nsb.Stop();	
-	    base.OnStop();
-	}
-}
-```
-
-Next to starting the role entry point, configure the endpoint behavior. In this case a hosting behavior, so that it will not run an endpoint itself but instead host other endpoints. To do so just specify the `AsA_Host` role.
-
-    public class EndpointConfig : IConfigureThisEndpoint, AsA_Host { }
+Snippet:HostingInWorkerRole
+
+Next to starting the role entry point, configure the endpoint behavior. In this case a hosting behavior, so that it will not run an endpoint itself but instead host other endpoints. To do so just specify the `AsA_Host` role for version 6 and below, or `IConfigureThisHost` for version 7 and up.
+
+Snippet:AsAHost
 
 The host entry point does require some configuration, it is necessary to tell it in what storage account to look for endpoints and how often it should do so, furthermore Azure needs to be configured to provision some space on the local disk, where the host can put the downloaded and extracted endpoints.
 
-Add the following configuration settings entries to the `.csdef` file
+For version 7 and above, this information is returned from the `Configure` method that must be implemented to satisfy the `IConfigureThisHost` interface.
```

Comment:
> For version 7 and above, this information is returned from the `Configure` method that must be implemented to satisfy the `IConfigureThisHost` interface.

For version 7 and above, return this information from the `Configure` method in the class implementing the `IConfigureThisHost` interface.